wallet/src/Pos/Util/Mnemonic.hs

```diff
@@ -0,0 +1,357 @@
+    arbitrary =
+        let
+            size    = fromIntegral $ natVal (Proxy @n)
+            entropy = mkEntropy  @n . B8.pack <$> vectorOf (size `quot` 8) arbitrary
+        in
+            either (error . show . UnexpectedMnemonicErr) identity <$> entropy
+
+
+-- Same remark from 'Arbitrary Entropy' applies here.
+instance
+    ( n ~ EntropySize mw
+    , mw ~ MnemonicWords n
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    , Arbitrary (Entropy n)
+    ) => Arbitrary (Mnemonic mw) where
+    arbitrary =
+        entropyToMnemonic <$> arbitrary @(Entropy n)
+
+
+instance Exception MnemonicException
+
+
+-- FIXME: Suggestion, we could -- when certain flags are turned on -- display
+-- a fingerprint of the Mnemonic, like a PKBDF2 over n iterations. This could be
+-- useful for debug to know whether two users are using the same mnemonic words
+-- and relatively benign EVEN THOUGH, it will permit to somewhat tight requests
+-- to a specific identity (since mnemonic words are 'unique', they are supposed
+-- to uniquely identify users, hence the privacy issue). For debbugging only and
+-- with the user consent, that's something we could do.
+instance Buildable (Mnemonic mw) where
+    build _ =
+        "<mnemonic>"
+
+instance Buildable (SecureLog (Mnemonic mw)) where
+    build _ =
+        "<mnemonic>"
+
+instance Buildable MnemonicErr where
+    build = \case
+        MnemonicErrInvalidEntropyLength l ->
+            bprint ("Entropy must be a sequence of " % build % " bytes") l
+        MnemonicErrFailedToCreate ->
+            bprint "Invalid Mnemonic words"
+        MnemonicErrForbiddenMnemonic ->
+            bprint "Forbidden Mnemonic: an example Mnemonic has been submitted. \
+            \Please generate a fresh and private Mnemonic from a trusted source"
+
+
+-- | To use everytime we need to show an example of a Mnemonic. This particular
+-- mnemonic is rejected to prevent users from using it on a real wallet.
+instance Default (Mnemonic 12) where
+    def =
+        let
+            wordsm =
+                [ "squirrel"
+                , "material"
+                , "silly"
+                , "twice"
+                , "direct"
+                , "slush"
+                , "pistol"
+                , "razor"
+                , "become"
+                , "junk"
+                , "kingdom"
+                , "flee"
+                ]
+
+            sentence = maybe
+                (error $ show $ UnexpectedMnemonicErr MnemonicErrFailedToCreate)
+                (mnemonicPhraseToMnemonicSentence Dictionary.english)
+                (mnemonicPhrase @12 (toUtf8String <$> wordsm))
+
+            entropy = fromMaybe
+                (error $ show $ UnexpectedMnemonicErr MnemonicErrFailedToCreate)
+                (wordsToEntropy @(EntropySize 12) sentence)
+        in Mnemonic
+            { mnemonicToSentence = sentence
+            , mnemonicToEntropy  = entropy
+            }
```

Comment:
Note, we cannot use `mkMnemonic` here because it uses this particular `Default` instance. This would create a mutual dependency in a way that won't converge.